src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -1050,9 +1051,21 @@ protected function isJsonCastable($key)
      */
     protected function isClassCastable($key)
     {
-        return array_key_exists($key, $this->getCasts()) &&
-                class_exists($class = $this->parseCasterClass($this->getCasts()[$key])) &&
-                ! in_array($class, static::$primitiveCastTypes);
+        if (! array_key_exists($key, $this->getCasts())) {
+            return false;
+        }
+
+        $castType = $this->parseCasterClass($this->getCasts()[$key]);
+
+        if (in_array($castType, static::$primitiveCastTypes)) {
+            return false;
+        }
+
+        if (class_exists($castType)) {
+            return true;
+        }
+
+        throw CastNotFoundException::make($this->getModel(), $key, $castType);
```

Comment:
This is confusing. It makes it sound like the exception happens when there's no cast. Maybe rename the class to `InvalidCastException` or `BadCastException`?